### PR TITLE
Implement local scope by adding overloads to the capture methods that accept a ScopeCallback

### LINKED ISSUE #1829: Implement local scope, which is a method parameter for the capture methods
Similar to Dart/Flutter https://github.com/getsentry/sentry-dart/pull/463/files

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description
Reopened because #2075 was auto closed by GitHub ...

Add overloads with scope callback for `captureEvent`, `captureException` and `captureMessage` to allow users to modify the scope for a single invocation of the capture methods.

## :bulb: Motivation and Context
fixes #1829 

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,15 @@
 # Changelog
 
+## Unreleased
+
+### Features
+
+- Implement local scope by adding overloads to the capture methods that accept a ScopeCallback ([#2075](https://github.com/getsentry/sentry-java/pull/2075))
```
Comment (adinauer) [inline CHANGELOG.md:7]:
```suggestion
- Implement local scope by adding overloads to the capture methods that accept a ScopeCallback ([#2084](https://github.com/getsentry/sentry-java/pull/2084))
```

**Comment by adinauer** [comment]
Opened https://github.com/getsentry/sentry-java/issues/2083 to deprecate `withScope`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+
 ## 6.0.0
 
 ### Sentry Self-hosted Compatibility
 
 - Starting with version `6.0.0` of the `sentry` package, [Sentry's self hosted version >= v21.9.0](https://github.com/getsentry/self-hosted/releases) is required or you have to manually disable sending client reports via the `sendClientReports` option. This only applies to self-hosted Sentry. If you are using [sentry.io](https://sentry.io), no action is needed.
 
 ### Features
-
```
Comment (marandaneto) [inline CHANGELOG.md:10]: I'd rather keep a line break after the headers

File: sentry/src/main/java/io/sentry/Hub.java [hunk 3]
```diff
@@ -761,4 +816,14 @@ SpanContext getSpanContext(final @NotNull Throwable throwable) {
     }
     return null;
   }
+
+  private Scope buildLocalScope(
+      final @NotNull Scope scope, final @Nullable ScopeCallback callback) {
+    if (callback != null) {
+      Scope localScope = new Scope(scope);
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/Hub.java:823]:
```suggestion
      final Scope localScope = new Scope(scope);
```

**Comment by marandaneto** [comment]
After releasing this, we can finally add local scope support for our Android support as well, this would require changing the docs for Java and Android, https://docs.sentry.io/platforms/android/enriching-events/scopes/ and https://docs.sentry.io/platforms/java/enriching-events/scopes/#local-scopes

**Review by marandaneto — APPROVED** [review]
@adinauer Thanks for doing this.
Left a few comments, but LGTM anyway so you are unblocked to fix and merge it.

File: CHANGELOG.md (hunk 2 above)
Comment (lbloder) [reply CHANGELOG.md]: Sorry, wasn't intentional :) new line is back

**Comment by adinauer** [comment]
> @adinauer Thanks for doing this.
@lbloder created the PR, i just reopened after GitHub auto closed it.

File: CHANGELOG.md (hunk 2 above)
Comment (adinauer) [reply CHANGELOG.md]: I think there were 2 before all the merging happened.

**Comment by adinauer** [comment]
Next step: add docs as suggested and keep `withScope` for now until it's deprecated.

> After releasing this, we can finally add local scope support for our Android support as well, this would require changing the docs for Java and Android, https://docs.sentry.io/platforms/android/enriching-events/scopes/ and https://docs.sentry.io/platforms/java/enriching-events/scopes/#local-scopes

**Pull request merged**